code/code/cmd/cmd_stab.cc

```diff
@@ -1,345 +1,239 @@
-
-      case WEAR_NECK:
-        if (tDamageType == SKILL_STABBING) {
-          tStringChar = "You slice $N's neck wide open!";
-          tStringVict = "$n slices your neck wide open!";
-          tStringOthr = "$n slices $N's neck wide open!";
-          tStringMess = "Blood spews forth from $N's severed jugular!";
-        } else {
-          tStringChar = "You sever $N at the neck!";
-          tStringVict = "$n severs you at the neck!";
-          tStringOthr = "$n severs $N at the neck!";
-          tStringMess = "Blood gushes from $N's neck!";
-        }
-
-        break;
-
-      case WEAR_BODY:
-        if (!(rand() % 9)) {
-          tStringChar = "You thrust $p DEEP into $N's gut!";
-          tStringVict =
-            "$n thusts $p DEEP into your gut, good thing you are dead now!";
-          tStringOthr = "$n thrusts $p DEEP into $N's gut!";
-        } else {
-          tStringChar =
-            "You plunge $p DEEP into $N's chest, you can feel $S heart slow "
-            "then stop through your weapon";
-          tStringVict =
-            "$n plunges $p DEEP into your chest, piercing your heart!";
-          tStringOthr =
-            "$n plunges $p DEEP into $N's chest, looks like he nailed them in "
-            "the heart!";
-          tStringMess = "Blood spews forth from the stab wound!";
-        }
-
-        break;
-
-      case WEAR_WAIST:
-        tStringChar = "You slice $N from love handle to love handle!";
-        tStringVict = "$n slices you from love handle to love handle!";
-        tStringOthr = "$n slices $N from love handle to love handle!";
-        break;
-
-      case WEAR_BACK:
-        if (!(rand() % 9) && !tSucker->raceHasNoBones()) {
-          tStringChar = "You rake $p down $N's back, slicing the spine!";
-          tStringVict = "$n rakes $p down your back, slicing your spine!";
-          tStringOthr = "$n rakes $p down the back, slicing their spine!";
-        } else {
-          tStringChar = "You carve a rather nice hole in $N's back!";
-          tStringVict = "$n carves you a good sized hole in your back!";
-          tStringOthr = "$n carves $N a good sized hole in the back!";
-        }
-
-        break;
-
-      default:
-        tStringChar = format(
-                        "Your stab to $N's %s ends their life in a final "
-                        "<r>spray of blood<z>!") %
-                      tStLimb;
-        tStringVict = format(
-                        "$n stabs you in your %s, ending your life in a final "
-                        "<r>spray of blood<z>!") %
-                      tStLimb;
-        tStringOthr = format(
-                        "$n stabs $N in $S %s, ending their life in a final "
-                        "<r>spray of blood<z>!") %
-                      tStLimb;
-        break;
-    }
+// Apply poison from weapon if present
+static void stabPoisonCheck(TBeing* victim, TGenWeapon* weapon) {
+  if (weapon->isPoisoned())
+    weapon->applyPoison(victim);
```

Comment:
It would be cool to move the `isPoisoned` into `applyPoison` itself (and update all call sites) - saves a dozen lines of code. Could make isPoisoned private, then.